kernel/src/actions/mod.rs

```diff
@@ -179,29 +179,43 @@ impl Protocol {
         })
     }
 
+    /// Create a new Protocol by visiting the EngineData and extracting the first protocol row into
+    /// a Protocol instance. If no protocol row is found, returns Ok(None).
     pub fn try_new_from_data(data: &dyn EngineData) -> DeltaResult<Option<Protocol>> {
         let mut visitor = ProtocolVisitor::default();
         data.extract(get_log_schema().project(&[PROTOCOL_NAME])?, &mut visitor)?;
         Ok(visitor.protocol)
     }
 
+    /// Get the minimum reader version for the protocol
```

Comment:
nit: "Get" is redundant for a getter (many style guides recommend not saying it)
```suggestion
    /// This protocol's minimum reader version
```
(more below)